### PR TITLE
Fix QQ Music url changed

### PR DESCRIPTION
**Pull request by gam2046**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [ ] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] New extractor
- [ ] Improvement
- [x] Bug fix
- [ ] New feature

---

### Description of your *pull request* and other information

Fix the url of `QQ Music`. like #13363

### CODE REVIEW

File: youtube_dl/extractor/qqmusic.py [hunk 1]
```diff
@@ -156,25 +156,37 @@ class QQPlaylistBaseIE(InfoExtractor):
     def qq_static_url(category, mid):
         return 'http://y.qq.com/y/static/%s/%s/%s/%s.html' % (category, mid[-2], mid[-1], mid)
 
-    @classmethod
-    def get_entries_from_page(cls, page):
+    def get_singer_all_songs(self, singmid, num):
+        return self._download_webpage(
+            r'https://c.y.qq.com/v8/fcg-bin/fcg_v8_singer_track_cp.fcg?format=json&inCharset=utf8&outCharset=utf-8&platform=yqq&needNewCode=0&singermid=%s&order=listen&begin=0&num=%s&songstatus=1' %
```
Comment (yan12125) [inline youtube_dl/extractor/qqmusic.py]: For long query strings, it's better to use ```query``` parameter here.

File: youtube_dl/extractor/qqmusic.py [hunk 2]
```diff
@@ -156,25 +156,37 @@ class QQPlaylistBaseIE(InfoExtractor):
...
         entries = []
 
-        for item in re.findall(r'class="data"[^<>]*>([^<>]+)</', page):
-            song_mid = unescapeHTML(item).split('|')[-5]
-            entries.append(cls.url_result(
-                'http://y.qq.com/#type=song&mid=' + song_mid, 'QQMusic',
-                song_mid))
+        default_num = 1
+        json_text = self.get_singer_all_songs(singmid, default_num)
+        json_obj = self._parse_json(json_text, singmid)
```
Comment (yan12125) [inline youtube_dl/extractor/qqmusic.py]: variable names like ```json_obj``` is not good. It doesn't describe what's the purpose of this variable.

File: youtube_dl/extractor/qqmusic.py [hunk 3]
```diff
@@ -156,25 +156,37 @@ class QQPlaylistBaseIE(InfoExtractor):
...
+
+        if json_obj['code'] == 0:
+            total = json_obj['data']['total']
+            json_text = self.get_singer_all_songs(singmid, total)
+            json_obj = self._parse_json(json_text, singmid)
+
+        for item in json_obj['data']['list']:
+            if not (item['musicData'].get('songmid') is None):
```
Comment (yan12125) [inline youtube_dl/extractor/qqmusic.py]: ```not (foo is None)``` => ```foo is not None```

File: youtube_dl/extractor/qqmusic.py [hunk 4]
```diff
@@ -292,7 +294,9 @@ class QQMusicToplistIE(QQPlaylistBaseIE):
     def _real_extract(self, url):
         list_id = self._match_id(url)
 
-        list_type, num_id = list_id.split("_")
+        # list_type, num_id = list_id.split("_")
```
Comment (yan12125) [inline youtube_dl/extractor/qqmusic.py]: Codes not used anymore can be just deleted.

File: youtube_dl/extractor/qqmusic.py [hunk 5]
```diff
@@ -352,7 +357,7 @@ def _real_extract(self, url):
         cdlist = list_json['cdlist'][0]
         entries = [
             self.url_result(
-                'http://y.qq.com/#type=song&mid=' + song['songmid'], 'QQMusic', song['songmid']
+                'https://y.qq.com/n/yqq/song/' + song['songmid'] + ".html", 'QQMusic', song['songmid']
```
Comment (yan12125) [inline youtube_dl/extractor/qqmusic.py]: In general we use single quotes unless there's a good reason. (single quotes in strings, etc.)

**Comment by gam2046** [comment]
@yan12125 Thanks, I'm new for Python. And the commit about `iqiyi` has been removed.

**Comment by yan12125** [comment]
iqiyi-related commits are still there. You may want to checkout “git rebase” to clean up commits.

**Comment by gam2046** [comment]
@yan12125 I'v rebase it. May I request an new PR ? There is no commit about `iqiyi` https://github.com/gam2046/youtube-dl/commits/master

**Comment by yan12125** [comment]
@gam2046: you can force push it: ```git push origin master --force``` (assume origin points to gam2046/youtube-dl)

**Comment by gam2046** [comment]
@yan12125 There is commit about `iqiyi`. https://github.com/rg3/youtube-dl/pull/13805/commits

**Comment by yan12125** [comment]
> @yan12125 There is commit about iqiyi. https://github.com/rg3/youtube-dl/pull/13805/commits

After force push? Then you'll need ```git rebase -i master``` and delete iqiyi lines.

**Comment by gam2046** [comment]
I did it (force push). Here is my `master` branch:
```
$ git log
commit e76cac918dde4da7db40b3d2c133341e9ae8a049
Author: forDream <gam2046@gmail.com>
Date:   Mon Aug 14 08:28:41 2017 +0800

    [qqmusic] review

commit 3ee7a3863bf068f941fdf7cff966d1bb382397d1
Author: forDream <gam2046@gmail.com>
Date:   Wed Aug 2 15:08:38 2017 +0800

    [qqmusic]support QQMusicSingerIE

commit d8d00305e41ffc8bc9761fc61103bd8e473545b4
Author: forDream <gam2046@gmail.com>
Date:   Wed Aug 2 12:20:53 2017 +0800

    [qqmusic]update valid url

commit 39091726a1c6116c48b9d3a8d3f4db6361c66ea4
Author: forDream <gam2046@gmail.com>
Date:   Wed Aug 2 11:12:17 2017 +0800

    fix QQ Music Url changed

commit 48d0c0772d83047470a65afc4e1c6ac8986dc471
Merge: 42820b719 9118c9f18
Author: forDream <gam2046@gmail.com>
Date:   Wed Aug 2 09:55:10 2017 +0800

    Merge remote-tracking branch 'refs/remotes/rg3/master'
```

**Comment by yan12125** [comment]
As you can see in this comparison: https://github.com/rg3/youtube-dl/compare/master...gam2046:master, there are old iqiyi-related commits. As ```git log``` sort commits by time, iqiyi-related commits do not appear in the first page.

And sorry for my previous incorrect instruction. As you're already on master, you'll need to rebase against upstream (rg3) master. Could you try this?
```
git remote add upstream https://github.com/rg3/youtube-dl
git fetch upstream master
git rebase -i upstream/master
```
Then yon can delete iqiyi commit from your favorite text editor.

**Comment by gam2046** [comment]
Did you means that there are two commits about `iqiyi` in 2015? 
I commit it before, and then changed it for myself, but not PR
I hava drop them, now.

**Comment by yan12125** [comment]
Thanks for the fix!

**Pull request merged**